File: split/heightplusdepth.pas
function heightplusdepth(f: internalfontnumber; c: quarterword): scaled;
var
  q: fourquarters;
  hd: eightbits;
begin
  q := fontinfo[charbase[f] + c].qqqq;
  hd := q.b1 - 0;
  heightplusdepth := fontinfo[heightbase[f] + (hd) div 16].int + fontinfo[depthbase[f] + (hd) mod 16].int;
end;

File: cpp/heightplusdepth.cpp
#include "heightplusdepth.h"

scaled heightplusdepth(internalfontnumber f, quarterword c)
{
	auto q = fontinfo[charbase[f]+c].qqqq;
	auto hd = q.b1;
	return fontinfo[heightbase[f]+hd/16].int_+fontinfo[depthbase[f]+hd%16].int_;
}
